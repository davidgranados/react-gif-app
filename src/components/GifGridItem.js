import React from 'react';

const GifGridItem = ({id, title, url}) => {
  return (
    <div id={id}>
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  );
};

export default GifGridItem;
