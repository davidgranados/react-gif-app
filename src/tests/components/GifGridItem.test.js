import React from "react";
import { shallow } from "enzyme";
import GifGridItem from "../../components/GifGridItem";

describe("Tests on <GifGridItem />", function () {
  const title = "a title";
  const url = "https://localhost.com/something.jpg";
  const wrapper = shallow(<GifGridItem id={"1"} title={title} url={url} />);

  it("should show the component correctly", function () {
    expect(wrapper).toMatchSnapshot();
  });

  it("should have a p with the title", function () {
    const p = wrapper.find("p");
    console.log(p.text());
    expect(p.text().trim()).toEqual(title);
  });

  it("should contains an img with the src and alt equals to the props url and title respectively", function () {
    const img = wrapper.find("img");
    console.log(img.html());
    console.log(img.props());
    console.log(img.prop("src"));
    console.log(img.prop("alt"));
    expect(img.prop("src")).toEqual(url);
    expect(img.prop("alt")).toEqual(title);
  });

  it('should have the card class', function () {
    const className = wrapper.prop("className");

    // multiple ways to do the same
    expect(className.includes("card")).toEqual(true);
    expect(wrapper.prop("className")).toContain("card");
    expect(wrapper.hasClass("card")).toEqual(true);
  });
});
