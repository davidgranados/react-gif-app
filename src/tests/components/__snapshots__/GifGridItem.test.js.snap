// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GifGridItem /> should show the component correctly 1`] = `
<div
  className="card animate__animated animate__bounce"
  id="1"
>
  <img
    alt="a title"
    src="https://localhost.com/something.jpg"
  />
  <p>
    a title
  </p>
</div>
`;
